### PR TITLE
fix(pytests) Clarify some tests and fixtures

### PR DESCRIPTION
**Pull request by Vince-janv**
## Description
As part of the pytest BBS series I made a few changes to some tests and fixtures

### Added
- assert-function to test content of file

### Changed

- Parametrization of post-processing test

### Fixed

- Name of fixture

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/meta/demultiplex/test_demux_post_processing.py [hunk 1]
```diff
@@ -32,23 +32,51 @@ def test_set_dry_run(
     assert post_demux_api.dry_run is True
 
 
+class Scenario(BaseModel):
```
Comment (henrikstranneheim) [inline tests/meta/demultiplex/test_demux_post_processing.py:35]:
```suggestion
class DemultiplexScenario(BaseModel):
```

File: tests/meta/demultiplex/test_demux_post_processing.py [hunk 2]
```diff
@@ -32,23 +32,51 @@ def test_set_dry_run(
...
+    flow_cell_directory: str
+    flow_cell_name: str
+    samples_ids: str
+
+
 @pytest.mark.parametrize(
-    "demux_type",
-    ["BCL2FASTQ_TREE", "BCLCONVERT_TREE", "BCLCONVERT_FLAT", "BCLCONVERT_ON_SEQUENCER"],
+    "scenario",
```
Comment (henrikstranneheim) [inline tests/meta/demultiplex/test_demux_post_processing.py:42]: ⭐ 
```suggestion
    "Demultiplex_scenario",
```

File: tests/meta/demultiplex/test_demux_post_processing.py [hunk 3]
```diff
@@ -120,6 +148,10 @@ def test_post_processing_of_flow_cell(
     assert delivery_path.exists()
 
 
+def assert_flow_cell_added_to_housekeeper(sample_ids: list[str]) -> None:
```
Comment (henrikstranneheim) [inline tests/meta/demultiplex/test_demux_post_processing.py:151]: How is this one used?

File: tests/meta/demultiplex/test_utils.py [hunk 4]
```diff
@@ -474,33 +474,38 @@ def test_create_manifest_file(tmp_flow_cells_directory_ready_for_demultiplexing_
...
     manifest_file: Path = create_manifest_file(
         tmp_flow_cells_directory_ready_for_demultiplexing_bcl_convert
     )
 
-    # THEN a manifest file should be created
+    # THEN a manifest manifest_file should be created
     assert manifest_file.exists()
 
-    # Then all files should be included in the manifest file
+    # Then all files should be included in the manifest manifest_file
```
Comment (henrikstranneheim) [inline tests/meta/demultiplex/test_utils.py:485]:
```suggestion
    # Then all files should be included in the manifest manifest file
```

**Review by henrikstranneheim — APPROVED** [review]
Nice 👍

File: tests/meta/demultiplex/test_demux_post_processing.py (hunk 3 above)
Comment (Vince-janv) [reply tests/meta/demultiplex/test_demux_post_processing.py:151]: Oops, leftover

File: tests/meta/demultiplex/test_demux_post_processing.py (hunk 2 above)
Comment (Vince-janv) [reply tests/meta/demultiplex/test_demux_post_processing.py:42]: Changed the name  of the class, but I would like to keep this as scenario for readability

File: tests/meta/demultiplex/test_utils.py [hunk 5]
```diff
@@ -32,10 +32,10 @@
 
 
 def test_validate_sample_fastq_with_valid_file():
-    # GIVEN a sample fastq file with a lane number and sample id in the parent directory name
+    # GIVEN a sample fastq manifest_file with a lane number and sample id in the parent directory name
```
[the remark this answers is not in the document]
Comment (Vince-janv) [reply tests/meta/demultiplex/test_utils.py:35]: Reformat went wrong, reverting it

File: tests/meta/demultiplex/test_demux_post_processing.py (hunk 2 above)
Comment (henrikstranneheim) [reply tests/meta/demultiplex/test_demux_post_processing.py:42]: 👍

**Pull request merged**